### ISSUE TITLE
Generate Error

### ISSUE THREAD

**Issue by xiewende** [dslisleedh/ESC#15]
I used model ESCRealM to train real-word task, the degradation was similar to that of Real-ESRGAN, but weaker.
The training process appears：

input｜output｜gt
<img width="1090" height="364" alt="Image" src="https://github.com/user-attachments/assets/6bf2cb10-0af9-48c3-a155-f6da79741df6" />

<img width="1092" height="362" alt="Image" src="https://github.com/user-attachments/assets/ff4eba48-fc8a-4936-9666-dacf626c5f2a" />

Why does this happen?

I'm confused. I haven't found the reason yet.

**Comment by dslisleedh** [comment]
Thanks for your interest, @xiewende ! 
Since I don’t know your exact training setup, here are a few tips that typically reduce training instability:

1. Pretrain MSE, then fine-tune GAN. Start with an L2/MSE model and switch to the GAN variant afterward.
2. Use a quarter-size pdim. Set pdim ≈ dim / 4 (e.g., `dim=64` --> `pdim=16`).
3. If you’re using dysample for upsampling, try a more stable upsampler such as NN+Conv.

You might also find the Enhance Everything Discord helpful. Related discussion: https://github.com/dslisleedh/ESC/issues/3#issuecomment-2719381744

If you can share the configuration you used, I’m happy to take a look.

**Comment by xiewende** [comment]
yeah, this error is occurred in MSE model
here is my training setup:

<img width="312" height="258" alt="Image" src="https://github.com/user-attachments/assets/2e94cea5-e1d9-4219-b7e4-212a042eb3ad" />

additionally，using nearest+conv for upsampling，just like [ESCRealM](https://github.com/dslisleedh/ESC/blob/main/esc/archs/esc_real_arch.py)

**Comment by dslisleedh** [comment]
I’m not 100% sure yet, but this might be related to the `unshuffle_mod` flag. 
When `upscaling_factor == 1`, it downsamples the input by x4 for computational efficiency.

So, if the training patch size is < 128 (and the corresponding feature size is <32), a single self-attention window will include padded values because `WindowAttention` pads H/W to multiples of the window size.
And, if the patch size is between 128 and 256, 3/4 of the windows include padded values, which can lead to training instability.

Could you try setting unshuffle_mod=False and training again?

Please let me know if this does not work, @xiewende !

**Comment by dslisleedh** [comment]
If the training patch is 256x256, `upscaling_factor` is 1, and `unshuffle_mod` is True, padding will definitely not occur, since the input feature is 64x64 and it is divisible by 32x32.

And, enabling `unshuffle_mod` does not mean you can't train other upscaling factors. 
I think Figure 4 in Real-ESRGAN would be clear to understand.

<img width="1349" height="398" alt="Image" src="https://github.com/user-attachments/assets/8508ba5b-6aa8-4720-a248-6bcaad934965" />

It just downsamples the input by x4 and x2 when the upscaling factor is x1 or x2, since this significantly accelerates inference speeds while dropping little performance.
So, you can train any upscaling factor, and there is an option to disable this.

But I think this might trigger the training instability since our network already leverages lots of operations that destabilize the training process (large kernel, partial channel design, and dynamic convolution).

Or maybe I'm wrong.
I wonder if setting `unshuffle_mod` to `False` would resolve this.

**Comment by xiewende** [comment]
yeah，u are right.
In my experience, there is some doubt about whether the operation pixelunshuffle will affect the stability of training.
But，I'm happy to try it（setting unshuffle_mod = false） and will give you feedback as soon as I get the results.

In addition, the reason I use upscaling_factor=1 is mainly to adapt to any size input and fixed size output. 
If the operation pixelunshuffle is not used, high inference speed is also a limitation.

**Comment by dslisleedh** [comment]
The specific reason is unclear, but there are two possible explanations.

First, unlike ESRGAN, which uses a pure CNN model, the linearly interpolating Transformer weights may not function properly.
For example, it's uncertain whether the blended `self-attention` operates as intended, and the blended gamma and beta in `LayerNorm` could cause significant distribution shifts.
So... you should probably verify if weight interpolation works correctly in other Transformer models.

Second, because a large kernel filter is shared across the entire model, even minor variations in this convolutional filter can cause significant distribution differences as layers deepen.

Hope you find a solution, @xiewende .

**Comment by xiewende** [comment]
yesh，
I agree with you。
due to the influence of attention and the kernel, subtle changes can also lead to huge differences in results. But it is difficult to clearly locate the specific direction

In addition, my original purpose of model fusion was to reduce the generation capacity.
Like the example below, after gan training, the generated data is unreasonable and dirty. 
Do you have any better methods to alleviate this phenomenon? @dslisleedh 

gan-before ｜ gan-after
<img width="924" height="342" alt="Image" src="https://github.com/user-attachments/assets/c1325023-4955-4857-9549-2f648fd60973" />
